### PR TITLE
2489 type error nil cant be coerced into fixnum

### LINKED ISSUE #2489: TypeError: nil can't be coerced into Fixnum
It seems there're problems with some users where  db_size_in_bytes returns nil.

https://rollbar.com/vizzuality/CartoDB/items/2730/?person_page=0#traceback

Error:
```
TypeError: nil can't be coerced into Fixnum
1
File "/home/ubuntu/www/production.cartodb.com/releases/20150225092349/app/models/user.rb", line 1172 in -
2
File "/home/ubuntu/www/production.cartodb.com/releases/20150225092349/app/models/user.rb", line 1172 in remaining_quota
3
File "/home/ubuntu/www/production.cartodb.com/releases/20150225092349/app/models/user/user_decorator.rb", line 27 in data
```

Other:
https://rollbar.com/vizzuality/CartoDB/items/4760/

```
TypeError: nil can't be coerced into Fixnum
1
File "/home/ubuntu/www/production.cartodb.com/releases/20150226104749/app/models/user.rb", line 1173 in -
2
File "/home/ubuntu/www/production.cartodb.com/releases/20150226104749/app/models/user.rb", line 1173 in remaining_quota
3
File "/home/ubuntu/www/production.cartodb.com/releases/20150226104749/services/importer/lib/importer/runner.rb", line 47 in initialize
4
File "/home/ubuntu/www/production.cartodb.com/releases/20150226104749/app/models/synchronization/member.rb", line 162 in new
5
File "/home/ubuntu/www/production.cartodb.com/releases/20150226104749/app/models/synchronization/member.rb", line 162 in run
6
File "/home/ubuntu/www/production.cartodb.com/releases/20150226104749/lib/resque/synchronization_jobs.rb", line 11 in block in perform
7
File "/home/ubuntu/www/production.cartodb.com/releases/20150226104749/lib/resque/base_job.rb", line 21 in call
8
File "/home/ubuntu/www/production.cartodb.com/releases/20150226104749/lib/resque/base_job.rb", line 21 in run_action
9
File "/home/ubuntu/www/production.cartodb.com/releases/20150226104749/lib/resque/synchronization_jobs.rb", line 10 in perform
```

### PR DESCRIPTION
**Pull request by juanignaciosl**
@Kartones I don't know what do you think about this attempt to fix #2489. That's a random error I see from time to time.

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1984/))

**Comment by Kartones** [comment]
uhmmm, I don't like having a custom statement timeout for a given call. We have a real problem calculating DB size and this will hide the problem instead of solving why a simple call to get some sizes takes too long.

In any case is my point of view, @javisantana what do you think?

**Comment by juanignaciosl** [comment]
Yep, `CDB_UserDataSize` is still a black box for me. Since you're working on those sizes we can halt this. Please let me know if you think I should dive into `CDB_UserDataSize` now.

**Comment by Kartones** [comment]
Well, I'm only changing the quota calculation functions. IIRC the big issue we have here is that once we introduced the raster tables to the calculation, times have grown up a lot, so it is no trivial issue.

Dunno really who will attack that, but again @javisantana is the one with precise answers.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2066/))

**Comment by juanignaciosl** [comment]
@Kartones , @javisantana check the logging proposal, please.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2067/))

**Comment by Kartones** [comment]
:+1:

File: app/models/user.rb [hunk 1]
```diff
@@ -1011,10 +1011,16 @@ def db_size_in_bytes
       user_data_size_function = self.cartodb_extension_version_pre_mu? ? "CDB_UserDataSize()" : "CDB_UserDataSize('#{self.database_schema}')"
       result = in_database(:as => :superuser).fetch("SELECT cartodb.#{user_data_size_function}").first[:cdb_userdatasize]
       result
-    rescue
+    rescue => e
       attempts += 1
-      in_database(:as => :superuser).fetch("ANALYZE")
+      begin
+        in_database(:as => :superuser).fetch("ANALYZE")
```
Comment (javisantana) [inline app/models/user.rb]: not sure about doing an analyze in a call that is supposed to be fast. This is not for this PR but to take into account in the future

**Comment by javisantana** [comment]
:es:

**Pull request merged**